src/libasr/pass/pass_manager.h

```diff
@@ -212,8 +216,8 @@ namespace LCompilers {
             _with_optimization_passes = {
                 "global_stmts",
                 "init_expr",
-                "implied_do_loops",
                 "class_constructor",
+                "implied_do_loops",
```

Comment:
This needs to be reverted. Sorry, I just noticed it while reviewing the sync PR at LFortran.